### ISSUE TITLE
Events for the programatic search

### ISSUE THREAD

**Issue by stephanrauh** [stephanrauh/ngx-extended-pdf-viewer#183]
Currently, `NgxExtendedPdfViewerService.find()` doesn't give you feedback. Search is asynchronous, so we'll have to add an optional callback.

- Easy goal: report the number of search result
- Difficult goal: return a list of search results. (Challenge: what is a meaningful result? Just an array of page numbers?)

**Comment by stephanrauh** [comment]
Instead of implementing a callback, I've added two events to `<ngx-extended-pdf-viewer>`:

```
<ngx-extended-pdf-viewer
  (updateFindMatchesCount)="updateFindMatchesCount($event)"
  (updateFindState)="updateFindState($event)">
```

**Comment by stephanrauh** [comment]
Shipped with version 1.8.2. There's also a live demo at https://pdfviewer.net/find.
